src/wallet/staking.cpp

```diff
@@ -478,11 +478,10 @@ bool CreateCoinStake(CWallet& wallet, unsigned int nBits, int64_t nSearchInterva
 
     // Sign
     int nIn = 0;
-    SignatureData empty;
 
     if (wallet.IsLegacy()) {
         for (const auto &pcoin : vwtxPrev) {
-            if (!SignSignature(*wallet.GetLegacyScriptPubKeyMan(), *pcoin, txNew, nIn++, SIGHASH_ALL, empty))
+            if (!SignSignature(*wallet.GetLegacyScriptPubKeyMan(), *pcoin, txNew, nIn++, SIGHASH_ALL))
```

Comment:
There is an easier and cleaner solution - moving the declaration of `empty` variable inside "for" loop, just before calling SignSignature().

Reply:
Looking through your code, couldnt see any need for the passing of the parameter,
However, for the sake of cleanness, have moved SignatureData empty inside of loop and removed placeholder functions